lldb/test/API/sanity/TestSettingSkipping.py

```diff
@@ -1,33 +1,33 @@
 """
-This is a sanity check that verifies that test can be sklipped based on settings.
+This is a sanity check that verifies that test can be skipped based on settings.
 """
 
-
 import lldb
 from lldbsuite.test.lldbtest import *
 from lldbsuite.test.decorators import *
 
 
 class SettingSkipSanityTestCase(TestBase):
     NO_DEBUG_INFO_TESTCASE = True
+    REAL_PYTHON_VERSION = "3.0"
+    FUTURE_PYTHON_VERSION = "4.0"
 
-    @skipIf(py_version=(">=", (3, 0)))
+    @skipIf(py_version=(">=", REAL_PYTHON_VERSION))
     def testSkip(self):
-        """This setting is on by default"""
-        self.assertTrue(False, "This test should not run!")
-
-    @skipIf(py_version=("<", (3, 0)))
-    def testNoMatch(self):
-        self.assertTrue(True, "This test should run!")
+        self.assertTrue(False, "This test should not run and fail (SKIPPED)")
 
-    @skipIf(setting=("target.i-made-this-one-up", "true"))
-    def testNotExisting(self):
-        self.assertTrue(True, "This test should run!")
+    @skipIf(py_version=("<", FUTURE_PYTHON_VERSION))
+    def testNoSKip(self):
+        self.assertTrue(True, "This test should run and pass(PASS)")
 
-    @expectedFailureAll(py_version=(">=", (3, 0)))
+    @expectedFailureAll(py_version=("<", FUTURE_PYTHON_VERSION))
     def testXFAIL(self):
-        self.assertTrue(False, "This test should run and fail!")
+        self.assertTrue(False, "This test should expectedly fail (XFAIL)")
 
-    @expectedFailureAll(py_version=("<", (3, 0)))
+    @expectedFailureAll(py_version=(">=", FUTURE_PYTHON_VERSION))
```

Comment:
is this right? what used to be <3.0 is now >=4.0?

Reply:
We had a bug in the decorator and I messed around with the conditions to test a few more cases, but the original ones were easier to understand. I'll keep the original comparisons.